scipy/optimize/_isotonic.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from ._pava_pybind import pava
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+__all__ = ["isotonic_regression"]
+
+
+def isotonic_regression(
+    y: npt.ArrayLike,
+    weights: npt.ArrayLike | None = None,
```

Comment:
As per https://docs.scipy.org/doc/scipy/dev/missing-bits.html#required-keyword-names
```suggestion
    y: npt.ArrayLike,
    *,
    weights: npt.ArrayLike | None = None,
```